### ISSUE TITLE
Issue: Threshold hesaplama mantığı hatası ve Metrik tutarsızlığı

### ISSUE THREAD

**Issue by EthanHunt45** [bugrakaann/AutoencoderLeatherDefectDetection#1]
Merhaba,

Projeyi ve notebook dosyasını inceledim iyi olmuş baya, eline sağlık. Ancak modelin performansını ve doğruluğunu etkileyen birkaç nokta var gibi:

1. **Threshold Mantığı:** Eşik değeri (threshold) şu anda `test_images` üzerinden hesaplanıyor. Test seti anomalileri de içerdiği için bu durum threshold değerini yukarı çekiyor ve gerçek hataların kaçırılmasına (False Negative) sebep olabiliyor. Bunun temiz eğitim seti üzerinden hesaplanması gerekir.
2. **Metrik Tutarsızlığı:** Model eğitimde `SSIM Loss` kullanırken, anomali tespiti aşamasında `MSE` kullanılıyor. README dosyasında da belirtildiği gibi doku (texture) tabanlı tespit için test aşamasında da SSIM farkı kullanılmalı. SSIM için doğrudan class oluşturdum.
3. **Legacy Format:** Model `.h5` formatında kaydediliyor, TensorFlow'un güncel `.keras` formatına geçilmesi daha sağlıklı olabilir.

Düzeltmelerin olduğu güncel dosya hazır. İnceleyip onaylaman için PR gönderiyorum.
